src/components/searcher/index.jsx

```diff
@@ -0,0 +1,30 @@
+import "./styles.css";
+
+const Searcher = ({ value, onChange, onSearch, placeholder }) => {
```

Comment:
I believe `SearchForm` is a more appropriate name for this component

Reply:
as I see from the task we need to pass 2 parameters: `initialValue` and `onSearch`. 
<img width="1192" alt="image" src="https://github.com/AnnaMartis/Epam-project/assets/36797396/935fc709-42ad-444d-8399-75645eec1ae8">


Just try to follow the task, it's better we do less but with better quality

Reply:
About this part, I didn't try to do more, I just didn't know how to do it differently.

I have a question. 

The requirement says 2 props - query and onSearch (which is a callback which fires when we want to submit an input).  But react threw an error ,when I used only this 2 props without onChange function and warned me that if I'm using controlled component I should also write onChange callback. (setQuery)

Thanks

Reply:
Well, let's figure it out 
1. onChange is a local state of the component, we need to manage inside (in out case), so it's basically temporary data that a user can type in input field. We can put it inside this component like that: 

const SearchForm = ({onSearch, initialValue }) =>{}

`const [searchValue, setSearchValue] = useState(initialValue);`

```
 const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

```

`onSearch` is something that we need to manage outside the component because this action can be different the this component should be responsible for that, so we can do it like that  

```
const handleFormSubmit = () =>{
onSearch(searchValue);
}
```

```
<form onSubmit={handleFormSubmit} > 
... 
```

please, let me know if it's not clear yet

Reply:
okaay,  I've got it fully.

Firstly I thought that the parent component has to have the searchValue state, but now I understand why we do like that. Because all we need to pass to a parent component is the final state which we want to submit. 
And we can pass it with onSearch function, so there is no need keeping searchValue state in parent component